server/src/main/java/org/elasticsearch/common/breaker/ChildMemoryCircuitBreaker.java

```diff
@@ -136,7 +136,7 @@ public double addEstimateBytesAndMaybeBreak(long bytes, String label) throws Cir
 
     @Override
     public long addBytesRangeAndMaybeBreak(long minAcceptableBytes, long wantedBytes, String label) throws CircuitBreakingException {
-        if (minAcceptableBytes == wantedBytes) {
+        if (minAcceptableBytes >= wantedBytes) {
```

Comment:
I'm not sure if this is the behavior we want? Maybe if `wantedBytes` is less than `minAcceptableBytes` we should fail?

Reply:
> Maybe if wantedBytes is less than minAcceptableBytes we should fail?

Sounds valid, otherwise it seems that the `minAcceptableBytes` intention is somehow by-passed.

Reply:
@seut pushed a fixup. 

Added a IllegalArgument case to the CircuitBreaker and raise the CircuitBreakingException outside. Felt like the more accurate separation of concerns. If the callee of `addBytesRangeAndMaybeBreak` needs less bytes than what it declares as acceptable, it is doing something wrong and needs to handle that case itself.